assets/reader-activation/index.js

```diff
@@ -199,6 +199,32 @@ export function getAuthStrategy() {
 	return getCookie( 'np_auth_strategy' );
 }
 
+/**
+ * Get a captcha token based on user input
+ */
+export async function getCaptchaToken( action = 'submit' ) {
+	return new Promise( ( res, rej ) => {
+		const { grecaptcha, newspack_reader_activation_data } = window;
+		if ( ! grecaptcha || ! newspack_reader_activation_data ) {
+			return res( '' );
+		}
+
+		const { captcha_site_key: captchaSiteKey } = newspack_reader_activation_data;
+		if ( ! grecaptcha?.ready || ! captchaSiteKey ) {
+			rej( 'Error loading the reCaptcha library.' );
+		}
+
+		grecaptcha.ready( async () => {
+			try {
+				const token = await grecaptcha.execute( captchaSiteKey, { action } );
+				return res( token );
+			} catch ( e ) {
+				rej( e );
+			}
+		} );
+	} );
+}
```

Comment:
I think we should either `async/await` or `Promise`s here.

Reply:
28a5869 refactors to remove the use of `async`/`await` on the front-end, which seems to be a problem for some browsers if not polyfilled.

Reply:
f497ba0 also fixes `auth.js`—the form field processing needs to happen in a `finally` statement in order to avoid a race condition between getting the reCAPTCHA token and submitting the form data.